### PR TITLE
chore: "compat" script for api compatibility check

### PR DESCRIPTION
**Pull request by eladb**
Add a `compat` script to all modules and to the repo root which runs api compatibility checks.
Also add a section in the contribution guide that explains about api checks.

Add a "build" and "pack" scripts in the root package.json so users can just run "yarn build/pack" from the root.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -479,6 +481,40 @@ run as part of the regular build, since they have some particular requirements.
...
+### API Compatibility Checks
+
+All stable APIs in the CDK go through a compatibility check during build using
+the [jsii-diff] tool. This tool downloads the latest released version from npm
+and verifies that the APIs in the current build have not changed in a breaking
+way.
+
+[jsii-diff]: https://www.npmjs.com/package/jsii-diff
+
+Compatibility checks always run as part of a full build (`build.sh`).
```
Comment (rix0rrr) [inline CONTRIBUTING.md]: Not `build.sh` then, but `yarn build`.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -479,6 +481,40 @@ run as part of the regular build, since they have some particular requirements.
...
+
+You can use `yarn compat` to run compatibility checks for all modules:
```
Comment (rix0rrr) [inline CONTRIBUTING.md]: This requires everything to have been built to jsii assemblies, might want to work that in there.

File: CONTRIBUTING.md (hunk 1 above)
Comment (eladb) [reply CONTRIBUTING.md]: fixed

File: CONTRIBUTING.md (hunk 2 above)
Comment (eladb) [reply CONTRIBUTING.md]: ok added "yarn build" before

**Pull request merged**